### PR TITLE
Remove unused stuff from mars.h

### PR DESCRIPTION
**Pull request by thewilsonator**
LDC still needs a [few things](https://github.com/ldc-developers/ldc/blob/master/dmd/mars.h#L98-L106) from mars.d so I haven't got rid of it entirely. Many of the thing in utils.d can now be `extern(D)`, I will follow up with another PR for that.

cc @kinke @ibuclaw 

None of this is used in the cxxfrontend tests.

### CODE REVIEW

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/mars.h [hunk 1]
```diff
@@ -10,87 +10,4 @@
...
-void obj_write_deferred(Library *library);
-
-/// Utility functions used by both main and frontend.
-FileBuffer readFile(Loc loc, const char *filename);
-void writeFile(Loc loc, const char *filename, const void *data, d_size_t size);
-void ensurePathToNameExists(Loc loc, const char *name);
-
-/// Little helper function for writing out deps.
-void escapePath(OutBuffer *buf, const char *fname);
+#include "arraytypes.h"
```
Comment (ibuclaw) [inline src/dmd/mars.h]: Then you might as well just remove mars.h in its entirety.

**Review by ibuclaw — DISMISSED** [review]

File: src/dmd/mars.h [hunk 2]
```diff
@@ -10,87 +10,4 @@
...
-        TARGET_OSX          Covers 32 and 64 bit Mac OSX
-        TARGET_FREEBSD      Covers 32 and 64 bit FreeBSD
-        TARGET_OPENBSD      Covers 32 and 64 bit OpenBSD
-        TARGET_DRAGONFLYBSD Covers 64 bit DragonFlyBSD
-        TARGET_SOLARIS      Covers 32 and 64 bit Solaris
-
-    It is expected that the compiler for each platform will be able
-    to generate 32 and 64 bit code from the same compiler binary.
-
-    There are currently no macros for byte endianness order.
```
Comment (ibuclaw) [inline src/dmd/mars.h]: Is this documentation anywhere else?

**Comment by WalterBright** [comment]
I'll defer .h file issues to the GDC and LDC teams, the only users of them.

File: src/dmd/mars.h (hunk 1 above)
Comment (thewilsonator) [reply src/dmd/mars.h]: Done.

**Pull request merged**